elementary/tracking/tracking_interface.py

```diff
@@ -27,46 +29,52 @@ def set_env(self, key: str, value):
         self._props[key] = value
 
     @abstractmethod
-    def _set_events_group(
-        group_type: str, group_identifier: str, group_props: Optional[dict] = None
+    def _register_group(
+        self, group_type: str, group_identifier: str, group_props: Optional[dict] = None
     ) -> None:
         raise NotImplementedError
 
     @abstractmethod
     def _send_event(
+        self,
         distinct_id: str,
         event_name: str,
         properties: Optional[dict] = None,
-        groups: Optional[dict] = None,
     ) -> None:
         raise NotImplementedError
 
+    def register_warehouse_group(self, warehouse_info: WarehouseInfo) -> None:
```

Comment:
I am not sure it should be here.
Not all of the tracking will need the register_warehouse_group  and this should be implemented only where needed (probably the anonymous tracking / outside of the tracking use)